backend/src/sms/services/sms-template.service.ts

```diff
@@ -226,11 +228,24 @@ const addToMessageLogs = async (
   }
 }
 
-const hasInvalidSmsRecipient = (
+/**
+ * Validate that recipient is a valid phone number and format it.
+ */
+const validateAndFormatNumber = (
   records: MessageBulkInsertInterface[]
-): boolean => {
-  const re = /^\+?[0-9]+$/
-  return records.some((record) => !re.test(record.recipient))
+): MessageBulkInsertInterface[] => {
+  return records.map((record) => {
```

Comment:
Hmm I get what you're doing here, not sure if it may end up being confusing at some point because the individual `record` objects already get mutated even if you don't return them (line 247) in this map function.

Reply:
Oh right, the unintended side effects could be confusing. Would it be better if I returned a new instance of record instead of modifying it in place?

Reply:
Yeah let's do that. 
The scenario I'm thinking of is:
```
let records = a
// validateAndFormatNumber gets invoked, mutating records
const formattedRecords = validateAndFormatNumber(records)

// I intended to use records as 'a' here, but it got mutated
someOtherFunction(records)
```